lib/connection.js

```diff
@@ -563,7 +565,7 @@ Connection.prototype.retrieve = function(type, ids, options, callback) {
     _.map(ids, function(id) {
       if (!id) { return Promise.reject(new Error('Invalid record ID. Specify valid record ID value')).thenCall(callback); }
       var url = [ self._baseUrl(), "sobjects", type, id ].join('/');
-      return self.request(url);
+      return self.request(url, options);
```

Comment:
> But I'd like to keep info in 'options' encapsulated and let request to handle 'options', just in case there is any info other than headers can be used in request.

https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it